store/src/main/java/org/apache/rocketmq/store/MappedFileQueue.java

```diff
@@ -421,7 +421,7 @@ public int deleteExpiredFileByOffset(long offset, int unitSize) {
 
     public boolean flush(final int flushLeastPages) {
         boolean result = true;
-        MappedFile mappedFile = this.findMappedFileByOffset(this.flushedWhere, false);
+        MappedFile mappedFile = this.findMappedFileByOffset(this.flushedWhere, this.flushedWhere == 0);
         if (mappedFile != null) {
```

Comment:
findMappedFileByOffset should keep its semantic restriction.
it is better to do like this:
MappedFile mappedFile = this.findMappedFileByOffset(this.flushedWhere, false);
if (mappedFile == null && this.flushedWhere == 0) {
   //revise the flush position, then find the file again
}